### PR TITLE
Update UniProt PTM xrefs to use CURIE syntax

### LINKED ISSUE #68: UniProt PTM references are not written as CURIEs
right now, UniProt post-translational modification references (https://bioregistry.io/registry/uniprot.ptm) are written as PTM-XXXX. 

This isn't CURIE syntax and doesn't make clear what the resource is that these come from. It would be good to switch them all to CURIE syntax using an appropriate prefix. I will send a PR to demonstrate.

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #68

This PR converts all references to UniProt PTMs that match the regular expression pattern `^PTM-\d{4}$` to have the `uniprot.ptm` prefix and be written in CURIE format.

The `uniprot.ptm` prefix is described at https://bioregistry.io/registry/uniprot.ptm. Other registries do not have this entry and as far as I can tell, UniProt has not done outreach to tell people what prefixes to use for their various resources.

### CODE REVIEW

**Comment by cthoyt** [comment]
Any updates on this? 

@pabinz @edeutsch

**Comment by edeutsch** [comment]
Thanks for the nudge. This seems ideal to me. I will try to get a decision very soon now.

**Review by edeutsch — APPROVED** [review]

**Review by mobiusklein — APPROVED** [review]
This seems likely to break existing readers that care about this field, but that's not a clear blocker here.

**Comment by althonos** [comment]
It seems the resulting CURIEs conflict with the format declared in Bioregistry: 
> Compact URIs (CURIEs) constructed from UniProt Post-Translational Modification should match this regular expression:
    `^uniprot\.ptm:\d{4}$`

In addition, you should declare an ID space in the OBO header so that the CURIE can be expanded by OBO readers:
```
idspace: uniprot.ptm https://bioregistry.io/uniprot.ptm: "UniProt Post-Translational Modification"
```
(this lets any OBO reader translate the CURIE to the Bioregistry URL by swapping the prefix with the given URL prefix).

**Comment by cthoyt** [comment]
@althonos let's talk about the issue with the redundant prefixes in the identifiers later. The Bioregistry has a field called the "banana' that helps rectify this. Obviously, the shape of the identifiers contains this redundant `PTM-` prefix within the _local unique identifiers_ which contains zero information content other than to compensate for LUIDs being used in place of actual CURIEs.

**Comment by cthoyt** [comment]
@pabinz @edeutsch @mobiusklein looks like everyone who read this is happy. Can we get a merge and a new release?

Also, I edited one document, but how do the changes get reflected in the others? Or will they get phased out?

**Comment by althonos** [comment]
@cthoyt : If you make your edits in the oldstyle OBO file, you can propagate the changes to the newstyle OBO and the OWL file with `make` (using `robot` and `fastobo` internally) :wink:

**Comment by cthoyt** [comment]
@althonos is the old style OBO file in a weird text encoding? any time I updated it in a text editor or with python the diff spans the whole file

**Comment by althonos** [comment]
I think the line breaks are Windows-style CR-LF :sweat_smile:

**Comment by cthoyt** [comment]
@althonos I've made the update, but it appears the `make` script heavily relies on the non-standard idiosyncracies, so i'll leave it to you to update that

**Comment by cthoyt** [comment]
@althonos scratch that, I think I figured it out

**Comment by cthoyt** [comment]
I just looked into the CI run and bumped the versions from `1.031.4` to `1.031.4` in each of the documents. Is there anything blocking merge now? It seems like all reviewers have said this is okay.  @pabinz @mobiusklein would one of you take care of a final check and merge please?

**Comment by edeutsch** [comment]
@cthoyt, this has now been merged, but @pabinz pointed out to me that it is not actually working. I believe that we agreed to change:
```
xref: PTM-0450
```
to:
```
xref: uniprot.ptm:PTM-0450
```

and that it what was committed.
But over at Bioregistry:
https://bioregistry.io/registry/uniprot.ptm
it still has the old style: 
xref: uniprot.ptm:0450
So I think that Bioregistry page needs to be updated to the agreed style.

PLUS, the example resolver URL on that page doesn't seem to work in either form:
https://bioregistry.io/reference/uniprot.ptm:0450
https://bioregistry.io/reference/uniprot.ptm:PTM-0450

Can this be fixed?

**Comment by cthoyt** [comment]
@edeutsch I appreciate your due diligence - we actually fixed this up earlier today: https://github.com/biopragmatics/bioregistry/pull/419. Tonight when it re-deploys, this will be fixed.

**Comment by cthoyt** [comment]
One other thing to mention is UniProt doesn't actually have a way to resolve these entries. I did get in touch to see if we could get this fixed - https://twitter.com/cthoyt/status/1510570256619778053 otherwise I will auto-generate an alternate site for this.

**Comment by edeutsch** [comment]
The primary uniprot.ptm page:
https://bioregistry.io/registry/uniprot.ptm
now shows correct CURIEs, but the example CURIE still does not work
https://bioregistry.io/reference/uniprot.ptm:PTM-0450

**Comment by cthoyt** [comment]
Hi @edeutsch, this page is a bit bare, which makes it hard to understand. There does not exist a website that shows information for a given UniProt PTM page, therefore the Bioregistry can not direct to one. Typically, even if this is the case, there might be alternate URIs constructed by other registries that this page would show.

I have come up with a remedy by auto-generating a site based on the UniProt source file. Right now it's available at https://biopragmatics.github.io/sites/uniprot.ptm/PTM-0001/ but I might change the URL pattern after thinking for a bit, then I will add to the bioregistry so at least it can redirect to something.

**Comment by edeutsch** [comment]
oh, I understand now, of course. And thanks for the temporary patch, looks good.

**Comment by cthoyt** [comment]
Screen Shot 2022-06-16 at 15 12 29

resolution is now enabled!

**Comment by edeutsch** [comment]
very nice, thank you!

**Pull request merged**
